DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/task_sequences/README.md

```diff
@@ -55,70 +55,109 @@ You can also find more information about [troubleshooting build errors](/DirectP
+| Invocation Capacity* | `uint32_t` | 1 | The size of the hardware queue instantiated for `async()` function calls. This parameter value corresponds to the minimum number of outstanding `async()` function calls to be supported. When the outstanding number of `async()` function calls reaches this value, further calls may block until the number of outstanding calls is reduced to the `invocation_capacity`. The default value of this parameter is 1.
+| Response Capacity* | `uint32_t` | 1 | The size of the hardware queue instantiated to hold task function results. This parameter value corresponds to the maximum number of outstanding `async()` calls such that all outstanding tasks are guaranteed to make forward progress. Further `async()` calls may block until the number of outstanding calls reduces to the `response_capacity`. The default value of this parameter is 1.
+*Invocation capacity and response capacity are optional 
 
-The following example shows how to use task Sequence.  
+The following example shows how to use task sequence.  
 
 ```c++
-// Task function
-int mult(int a, int b) {
-	 return a * b;
+[[intel::use_stall_enable_clusters]] 
+void LoopA(int len) {
+  [[intel::initiation_interval(1)]]  
+  for (size_t i = 0; i < len; i++) {
+    int in0 = PipeIn0::read();
+    int in1 = PipeIn1::read();
+
+    PipeAB::write(in0);
+    PipeAD::write(in1);
+  }
 }
 
-int mult_and_add(int a1, int b1, int a2, int b2) {
-  task_sequence<mult> task1, task2;
-  task1.async(a1, b1);
-  task2.async(a2, b2);
-  return task1.get() + task2.get();
+[[intel::use_stall_enable_clusters]] 
+void LoopB(int len) {
+  [[intel::initiation_interval(1)]]  
+  for (size_t i = 0; i < len; i++) {
+    int tmp = PipeAB::read();
+    tmp += i;
+    PipeBC::write(tmp);  
+  }
 }
-```
-C++ function, mult is the task function with asynchronous activity defined. The task function is used to parameterize the task_sequence class with two object instances, task1 and task2. Parallel invocations of the task function is then launched through async() method of each object instance and their results collected by calls to the get() method on each object. The
-async() function call is non-blocking, and it returns before the asynchronous f invocation completes executing and potentially before f even begins executing, as the return type from the async() function provides no implicit information on the execution status of f.
-The get() function retrieves the oldest result from this logical FIFO queue and blocks (waits) until a result is available if no result is available immediately upon the call to the get() function. The return type of the get() function is the same as the return type of the task function .
 
-You can see concrete example of kernel that use task sequences to run 4 loops concurrently in `task_sequence/main.cpp` and naively running the 4 loops sequentially in naive method in `naive/main.cpp`.
+[[intel::use_stall_enable_clusters]] 
```

Comment:
This makes it look like `[[intel::use_stall_enable_clusters]] is specifically necessary for `task_sequence`? I recall that we added it because the compiler was adding too much capacity in between the `async()` calls. If that bug has not been fixed for 2024.1, please make a note in the HSD ES case that these attributes should be removed when the bug is fixed.

also make a comment in the source code that the `[[intel::use_stall_enable_clusters]]` attribute is there to work around a compiler bug that hurts performance.

You can start your search for the HSD ES case here (this query looks for cases with CEG_HLD_ONEAPI tag): https://hsdes.intel.com/appstore/community/#/1408282547?queryId=18031042003